lib/flatpak-installation.c

```diff
@@ -337,16 +337,38 @@ flatpak_installation_new_for_path (GFile *path, gboolean user,
 }
 
 static FlatpakDir *
-flatpak_installation_get_dir (FlatpakInstallation *self)
+_flatpak_installation_get_dir (FlatpakInstallation *self, gboolean allow_no_repo)
 {
   FlatpakInstallationPrivate *priv = flatpak_installation_get_instance_private (self);
   FlatpakDir *dir;
   G_LOCK (dir);
   dir = g_object_ref (priv->dir_unlocked);
+  if (!allow_no_repo && flatpak_dir_get_repo (dir) == NULL)
+    {
+      g_autoptr(GError) local_error = NULL;
+      if (!flatpak_dir_ensure_repo (dir, NULL, &local_error))
+        {
+          g_autofree char *dir_name = flatpak_dir_get_name (dir);
+          g_warning ("Unable to ensure repo for %s: %s",
```

Comment:
You need to add a GError to get_dir() and pass the error on to callers.

Reply:
Otherwise things will start asserting when there is no repo.

Reply:
Fixed.